develop/app/Controllers/v2/PaymentController.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+namespace App\Controllers\v2;
+
+use CodeIgniter\API\ResponseTrait;
+
+use App\Controllers\BaseController;
+use App\Models\v2\PaymentModel;
+use App\Entities\v2\PaymentEntity;
+use App\Models\v2\WalletModel;
+use App\Services\UserService;
+
+class PaymentController extends BaseController
+{
+    use ResponseTrait;
+
+    private $u_key;
+
+    public function __construct()
+    {
+        $this->u_key = UserService::getUserKey();
+    }
+
+    /**
+     * [GET] /api/v1/payment
+     * Get payment data.
+     *
+     * @return void
+     */
+    public function index()
+    {
+        $limit  = $this->request->getGet("limit")  ?? 10;
+        $offset = $this->request->getGet("offset") ?? 0;
+        $search = $this->request->getGet("search") ?? 0;
+        $isDesc = $this->request->getGet("isDesc") ?? "desc";
+        $u_key  = $this->u_key;
+
+        $paymentModel  = new PaymentModel();
+        $paymentEntity = new PaymentEntity();
+
+        $query = $paymentModel->orderBy("created_at", $isDesc ? "DESC" : "ASC");
+        if ($search !== 0) {
+            $query->like("o_key", $search);
+        }
+        $dataCount = $query->countAllResults(false);
+        $payments  = $query->where("u_key", $u_key)->findAll($limit, $offset);
+
+        $data = [
+            "list"      => [],
+            "dataCount" => $dataCount
+        ];
+
+        if ($payments) {
+            foreach ($payments as $paymentEntity) {
+                $paymentData = [
+                    "pm_key" => $paymentEntity->pm_key,
+                    "u_key"  => $paymentEntity->u_key,
+                    "o_key"  => $paymentEntity->o_key,
+                    "status" => $paymentEntity->status,
+                    "total"  => $paymentEntity->total
+                ];
+                $data["list"][] = $paymentData;
+            }
+        } else {
+            return $this->fail("Payment data not found", 404);
+        }
+
+        return $this->respond([
+            "status" => true,
+            "data"   => $data,
+            "msg"    => "Payment index method successful",
+        ]);
+    }
+
+    /**
+     * [GET] /api/v1/payment/{paymentKey}
+     * Get payment data by payment key.
+     *
+     * @param int $paymentKey
+     * @return void
+     */
+    public function show($paymentKey = null)
+    {
+        if ($paymentKey === null) {
+            return $this->fail("The payment key is required.", 404);
+        }
+
+        $paymentModel = new PaymentModel();
+
+        $paymentEntity = $this->getPaymentEntityByPaymentKey($paymentKey);
```

Comment:
This method name is too long, may use `getPaymentEntity`.

Reply:
This change is done.